data_structures/external_memory_node.cpp

```diff
@@ -40,13 +40,13 @@ ExternalMemoryNode::ExternalMemoryNode() : barrier(false), traffic_lights(false)
 
 ExternalMemoryNode ExternalMemoryNode::min_value()
 {
-    return ExternalMemoryNode(0, 0, 0, false, false);
+    return ExternalMemoryNode(0, 0, NodeID64(0), false, false);
 }
 
 ExternalMemoryNode ExternalMemoryNode::max_value()
 {
     return ExternalMemoryNode(std::numeric_limits<int>::max(), std::numeric_limits<int>::max(),
-                              std::numeric_limits<unsigned>::max(), false, false);
+                              NodeID64(std::numeric_limits<unsigned>::max()), false, false);
```

Comment:
This looks strange. Why is max of `unsigned` used here? Shouldn't this be max of whatever is inside the `NodeID64`? Maybe have a `static const NodeID64::max_value;` in the `NodeID64` or something like that?